aptos-move/framework/src/extended_checks.rs

```diff
@@ -154,26 +154,50 @@ impl<'a> ExtendedChecker<'a> {
                     &format!("`{}` function must be private", INIT_MODULE_FUN),
                 )
             }
-            for Parameter(_, ty, _) in fun.get_parameters() {
+
+            if fun.get_parameter_count() != 1 {
+                self.env.error(
+                    &fun.get_id_loc(),
+                    &format!(
+                        "`{}` function can only take a single value of type `signer` \
+                         or `&signer` as parameter",
+                        INIT_MODULE_FUN
+                    ),
+                )
+            } else {
+                let Parameter(_, ty, _) = &fun.get_parameters()[0];
                 let ok = match ty {
                     Type::Primitive(PrimitiveType::Signer) => true,
-                    Type::Reference(_, ty) => matches!(*ty, Type::Primitive(PrimitiveType::Signer)),
+                    Type::Reference(_, ty) => {
+                        matches!(ty.as_ref(), Type::Primitive(PrimitiveType::Signer))
+                    },
                     _ => false,
                 };
                 if !ok {
                     self.env.error(
                         &fun.get_id_loc(),
-                        &format!("`{}` function can only take values of type `signer` or `&signer` as parameters",
-                                 INIT_MODULE_FUN),
+                        &format!(
+                            "`{}` function can only take an argument of type `signer` \
```

Comment:
nit: rewrite to - "`{}` function can only take a single parameter of type `signer` or `&signer`" (perhaps the same for the error message above for != 1 case)

Reply:
Done: just shared the same error message for both.